lib/dialects/mysql/query/mysql-querycompiler.js

```diff
@@ -9,12 +9,13 @@ class QueryCompiler_MySQL extends QueryCompiler {
     super(client, builder, formatter);
 
     const { returning } = this.single;
-    if (returning) {
+    if (returning && (!client.version || parseFloat(client.version) < 10.5)) {
```

Comment:
I would extract the version check in a separate function and write tests for it. We need to support both the `10.5` and the `10.6.3` format, and I don't think that current solution does.
In order to avoid reparsing on every query, I think we could check it once when creating a client instance, and then set a flag `isReturningSupported` that will be reused for checks.